File: backend/app/schemas/comment.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class CommentBase(BaseModel):
    body: str = Field(..., min_length=1, max_length=2000)


class CommentCreate(CommentBase):
    ...


class CommentUpdate(BaseModel):
    body: str = Field(..., min_length=1, max_length=2000)


class CommentAuthor(BaseModel):
    id: int
    username: str

    model_config = ConfigDict(from_attributes=True)


class CommentRead(CommentBase):
    id: int
    project_id: int
    author: CommentAuthor
    edited: bool
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/comment.ts
export type ProjectCommentAuthor = {
  id: number;
  username: string;
};

export type ProjectComment = {
  id: number;
  project_id: number;
  body: string;
  edited: boolean;
  created_at: string;
  updated_at: string;
  author: ProjectCommentAuthor;
};
